### ISSUE TITLE
squared hinge loss code

### ISSUE THREAD

**Issue by tdhock** [lamtung16/ML_ChangepointDetection#3]
for torch learning of penalty function please use squared hinge loss with margin 1 as below,
```python
import torch
pred_scores =torch.tensor([-5, -3.5, -2.5, -1.5, 0])
def positive_part(x):
    return torch.where(x>0, x, 0)
def squared_hinge_loss(pred_log_penalty, lower_limit, upper_limit, margin=1):
    lower_margin = lower_limit+margin
    upper_margin = upper_limit-margin
    lower_diff = positive_part(lower_margin-pred_log_penalty)
    upper_diff = positive_part(pred_log_penalty-upper_margin)
    total_diff = lower_diff + upper_diff
    return (total_diff ** 2).mean()
loss_value = squared_hinge_loss(pred_scores, -4, -1)
loss_value.backward()
```
